leeedohyun/src/test/java/inflearn/mini/annualleave/controller/AnnualLeaveControllerTest.java

```diff
@@ -0,0 +1,86 @@
+import static org.springframework.test.web.servlet.request.MockMvcRequestBuilders.post;
+import static org.springframework.test.web.servlet.result.MockMvcResultMatchers.status;
+
+import java.time.LocalDate;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.web.servlet.WebMvcTest;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.http.MediaType;
+import org.springframework.test.web.servlet.MockMvc;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+
+import inflearn.mini.annualleave.dto.request.AnnualLeaveRequestDto;
+import inflearn.mini.annualleave.dto.request.RemainingAnnualLeaveRequestDto;
+import inflearn.mini.annualleave.dto.response.RemainingAnnualLeaveResponseDto;
+import inflearn.mini.annualleave.exception.InvalidAnnualLeaveRequestException;
+import inflearn.mini.annualleave.service.AnnualLeaveService;
+
+@WebMvcTest(AnnualLeaveController.class)
+class AnnualLeaveControllerTest {
+
+    @Autowired
+    private MockMvc mockMvc;
+
+    @Autowired
+    private ObjectMapper objectMapper;
+
+    @MockBean
+    private AnnualLeaveService annualLeaveService;
+
+    @Test
+    void 연차를_신청한다() throws Exception {
+        // given
+        final AnnualLeaveRequestDto request = new AnnualLeaveRequestDto(1L, LocalDate.of(2024, 3, 15));
+
+        // when
+        doNothing().when(annualLeaveService).requestAnnualLeave(any());
+
+        // then
+        mockMvc.perform(post("/api/v1/annual-leaves")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(request)))
+                .andExpect(status().isOk());
+    }
+
+    @Test
+    void 연차_신청에_실패한다() throws Exception {
+        // given
+        final AnnualLeaveRequestDto request = new AnnualLeaveRequestDto(1L, LocalDate.of(2024, 3, 15));
+
+        // when
+        doThrow(InvalidAnnualLeaveRequestException.class).when(annualLeaveService).requestAnnualLeave(any());
+
+        // then
+        mockMvc.perform(post("/api/v1/annual-leaves")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(request)))
+                .andExpect(status().isBadRequest());
+    }
+
+    @Test
+    void 남은_연차_조회한다() throws Exception {
+        // given
+        final RemainingAnnualLeaveRequestDto request = new RemainingAnnualLeaveRequestDto(1L);
+
+        given(annualLeaveService.getRemainingAnnualLeave(any()))
+                .willReturn(new RemainingAnnualLeaveResponseDto(15));
+
+        // when
+
+        // then
+        mockMvc.perform(get("/api/v1/annual-leaves")
+                        .contentType(MediaType.APPLICATION_JSON)
+                        .content(objectMapper.writeValueAsString(request)))
+                .andExpect(status().isOk());
+    }
```

Comment:
```suggestion
        // when & then
        mockMvc.perform(get("/api/v1/annual-leaves")
                        .contentType(MediaType.APPLICATION_JSON)
                        .content(objectMapper.writeValueAsString(request)))
                .andExpect(status().isOk());
    }
```
`when` 과 `then` 이 함께 있는경우가 종종 있어 주석을 붙이는게 애매할때가 있습니다.
저는 이럴때 `// when & then` 이런식으로 표기하기도 하는데
좀 더 좋은 표기법이 있는지 고민해 보는것도 좋을것 같습니다 ㅎㅎ

Reply:
맞아요! 이럴 때 어떻게 표기하는게 좋을지 모르겠어요! 이번 기회에 고민해보겠습니다